minidump-processor/src/processor.rs

```diff
@@ -256,7 +256,7 @@ where
         let stack = thread.stack_memory(&memory_list);
 
         let mut stack =
-            stackwalker::walk_stack(&context, stack.as_deref(), &modules, symbol_provider);
+            stackwalker::walk_stack(&context, stack.as_deref(), &modules, symbol_provider).await;
```

Comment:
I guess if you really wanted to go all-in here you could map the threads to individual `async {}` blocks and use [`future::join_all`](https://docs.rs/futures/latest/futures/future/fn.join_all.html) to get all the threads? Might require some rejiggering of the symbol stuff to actually work properly, like if one thread requests a symbol file and is waiting on it then another thread requests the same file before it's done it'd be great to return it a future that also resolves when the file is finished downloading and ready to go. Still probably overkill since everything would synchronize around downloading symbols for libxul no matter what you do. :shrug:

Reply:
Yeah I think the place where you might actually get some interesting advantage is if you could somehow multiplex(??) downloading the symbol and writing it into the cache, so that you don't block on writing a 1GB file to the cache after spending all that time downloading a 1GB file.

This is of course assuming there isn't a shared bottleneck between disk i/o and network i/o (no fucking clue). And also I just have no idea how to express that!

Reply:
Not going to think about it too hard since this is in service of making it easier to swap in symbolic which would presumably handle interesting optimizations for us.